packages/strapi-plugin-users-permissions/services/UsersPermissions.js

```diff
@@ -0,0 +1,250 @@
+      return acc;
+    }, []).sort()) + 1;
+
+    const newRole = _.pick(role, ['name', 'description', 'permissions']);
+
+    _.set(appRoles, highestId.toString(), newRole);
+
+    _.forEach(role.users, (user) => {
+      Service.updateUserRole(user, highestId);
+    });
+
+    Service.writePermissions(appRoles);
+  },
+
+  deleteRole: async (roleId) => {
+    const Service = strapi.plugins['users-permissions'].services.userspermissions;
+    const appRoles = require(Service.getRoleConfigPath());
+
+    Service.writePermissions(_.omit(appRoles, [roleId]));
+
+    const users = await strapi.query('user', 'users-permissions').find(strapi.utils.models.convertParams('user', {
+      role: roleId
+    }));
+
+    _.forEach(users, (user) => {
+      Service.updateUserRole(user, '1');
+    });
+  },
+
+  getActions: () => {
+    const generateActions = (data) => (
+      Object.keys(data).reduce((acc, key) => {
+        acc[key] = { enabled: false, policy: '' };
+
+        return acc;
+    }, {}));
+
+    const appControllers = Object.keys(strapi.api || {}).reduce((acc, key) => {
+      acc.controllers[key] = generateActions(strapi.api[key].controllers[key]);
+
+      return acc;
+    }, { controllers: {} });
+
+    const pluginsPermissions = Object.keys(strapi.plugins).reduce((acc, key) => {
+      acc[key] = Object.keys(strapi.plugins[key].controllers).reduce((obj, k) => {
+        obj.controllers[k] = generateActions(strapi.plugins[key].controllers[k]);
+
+        return obj;
+
+      }, { controllers: {} });
+
+      return acc;
+    }, {});
+
+    const permissions = {
+      application: {
+        controllers: appControllers.controllers,
+      },
+    };
+
+    const allPermissions = _.merge(permissions, pluginsPermissions);
+
+    return allPermissions;
+  },
+
+  getRole: async (roleId) => {
+    const Service = strapi.plugins['users-permissions'].services.userspermissions;
+    const appRoles = require(Service.getRoleConfigPath());
+    appRoles[roleId].users = await strapi.query('user', 'users-permissions').find(strapi.utils.models.convertParams('user', { role: roleId }));
+
+    return appRoles[roleId];
+  },
+
+  getRoles: async () => {
+    const Service = strapi.plugins['users-permissions'].services.userspermissions;
+    const roles = require(Service.getRoleConfigPath());
+    const usersCount = await strapi.query('user', 'users-permissions').countByRoles();
+    const formattedRoles = Object.keys(roles).reduce((acc, key) => {
+      const role = _.pick(roles[key], ['name', 'description']);
+
+      _.set(role, 'id', key);
+      _.set(role, 'nb_users', _.get(_.find(usersCount, { _id: parseFloat(key) }), 'total', 0));
+      acc.push(role);
+
+      return acc;
+    }, []);
+
+    return formattedRoles;
+  },
+
+  getRoutes: async () => {
+    return Object.keys(strapi.plugins).reduce((acc, current) => {
+      acc[current] = strapi.plugins[current].config.routes;
+
+      return acc;
+    }, {});
+  },
+
+  getRoleConfigPath: () => (
```

Comment:
The file is already loaded so if you apply my changes, you don't need it anymore